src/libraries/Common/src/Interop/Browser/Interop.JavaScript.TypedArray.cs

```diff
@@ -0,0 +1,285 @@
+            //         case Float32Array _:
+            //             return TypedArrayTypeCode.Float32Array;
+            //         case Float64Array _:
+            //             return TypedArrayTypeCode.Float64Array;
+            //         default:
+            //             throw new ArrayTypeMismatchException("TypedArray is not of correct type.");
+            //     }
+            // }
+
+            public int BytesPerElement => (int)GetObjectProperty("BYTES_PER_ELEMENT");
+            public string Name => (string)GetObjectProperty("name");
+            public int ByteLength => (int)GetObjectProperty("byteLength");
+            public ArrayBuffer Buffer => (ArrayBuffer)GetObjectProperty("buffer");
+
+            public void Fill(U value) => Invoke("fill", value);
+            public void Fill(U value, int start) => Invoke("fill", value, start);
+            public void Fill(U value, int start, int end) => Invoke("fill", value, start, end);
+
+            public void Set(Array array) => Invoke("set", array);
+            public void Set(Array array, int offset) => Invoke("set", array, offset);
+            public void Set(ITypedArray typedArray) => Invoke("set", typedArray);
+            public void Set(ITypedArray typedArray, int offset) => Invoke("set", typedArray, offset);
+
+            public T Slice() => (T)Invoke("slice");
+            public T Slice(int begin) => (T)Invoke("slice", begin);
+            public T Slice(int begin, int end) => (T)Invoke("slice", begin, end);
+
+            public T SubArray() => (T)Invoke("subarray");
+            public T SubArray(int begin) => (T)Invoke("subarray", begin);
+            public T SubArray(int begin, int end) => (T)Invoke("subarray", begin, end);
+
+            public U? this[int i]
+            {
+                get
+                {
+                    var jsValue = Runtime.GetByIndex(JSHandle, i, out int exception);
+
+                    if (exception != 0)
+                        throw new JSException((string)jsValue);
+
+                    // The value returned from the index.
+                    return UnBoxValue(jsValue);
+                }
+                set
+                {
+                    object res = Runtime.SetByIndex(JSHandle, i, value, out int exception);
+
+                    if (exception != 0)
+                        throw new JSException((string)res);
+
+                }
+            }
+
+            private U? UnBoxValue(object jsValue)
+            {
+                if (jsValue != null)
+                {
+                    var type = jsValue.GetType();
+                    if (type.IsPrimitive)
+                    {
+                        return (U)Convert.ChangeType(jsValue, typeof(U));
+                    }
+                    else
+                    {
+                        throw new InvalidCastException($"Unable to cast object of type {type} to type {typeof(U)}.");
```

Comment:
Isn't this what Convert.ChangeType does throw?

Reply:
Addressed in recent commits